src/WordPress/Blueprints/Runner/Step/CpStepRunner.php

```diff
@@ -2,18 +2,53 @@
 
 namespace WordPress\Blueprints\Runner\Step;
 
+use Symfony\Component\Filesystem\Exception\IOException;
+use Symfony\Component\Filesystem\Filesystem;
+use WordPress\Blueprints\BlueprintException;
 use WordPress\Blueprints\Model\DataClass\CpStep;
 
+class CpStepRunner extends BaseStepRunner
+{
+    /**
+     * @param CpStep $input
+     */
+    public function run(CpStep $input)
+    {
+        $resolvedFromPath = $this->getRuntime()->resolvePath($input->fromPath);
+        $resolvedToPath = $this->getRuntime()->resolvePath($input->toPath);
 
-class CpStepRunner extends BaseStepRunner {
-	/**
-	 * @param CpStep $input
-	 */
-	function run( CpStep $input ) {
-		// @TODO: Treat the input paths as relative path to the document root (unless it's absolute)
-		$success = copy( $input->fromPath, $input->toPath );
-		if ( ! $success ) {
-			throw new \Exception( "Failed to copy file from {$input->fromPath} to {$input->toPath}" );
-		}
-	}
+        $fileSystem = new Filesystem();
+
+        try {
+            // TODO affirm it should overwrite newer files
+            // Filesystem's copy requires flag `overwriteNewerFiles` set to `true`
```

Comment:
What are "newer" files? Is the timestamp compared?

Reply:
Yes. Excerpt from the Filesystem docs:

> If the target file is older than the origin file, it's always overwritten. If the target file is newer, it is overwritten only when the $overwriteNewerFiles option is set to true.

I'd assume the steps should just do as they are told. Therefore, this suggestion.

Reply:
yeah, let's ignore the timestamps and just overwrite any pre-existing files